authentication/oauth_test.go

```diff
@@ -18,15 +18,34 @@ import (
 )
 
 func TestOAuthLoginWithPassword(t *testing.T) {
-	configureHTTPTestRecordings(t)
+	t.Run("Should return tokens", func(t *testing.T) {
+		configureHTTPTestRecordings(t)
 
-	tokenSet, err := authAPI.OAuth.LoginWithPassword(context.Background(), oauth.LoginWithPasswordRequest{
-		Username: "testuser",
-		Password: "testuser123",
-	}, oauth.IDTokenValidationOptions{})
-	assert.NoError(t, err)
-	assert.NotEmpty(t, tokenSet.AccessToken)
-	assert.Equal(t, "Bearer", tokenSet.TokenType)
+		tokenSet, err := authAPI.OAuth.LoginWithPassword(context.Background(), oauth.LoginWithPasswordRequest{
+			Username: "testuser",
+			Password: "testuser123",
+		}, oauth.IDTokenValidationOptions{})
+		assert.NoError(t, err)
+		assert.NotEmpty(t, tokenSet.AccessToken)
+		assert.Equal(t, "Bearer", tokenSet.TokenType)
+	})
+
+	t.Run("Should support passing extra options", func(t *testing.T) {
+		configureHTTPTestRecordings(t)
+
+		tokenSet, err := authAPI.OAuth.LoginWithPassword(context.Background(), oauth.LoginWithPasswordRequest{
+			Username: "testuser",
+			Password: "testuser123",
+			Realm:    "my-realm",
+			Scope:    "extra-scope",
+			ExtraParameters: map[string]string{
+				"extra": "value",
+			},
+		}, oauth.IDTokenValidationOptions{})
+		assert.NoError(t, err)
+		assert.NotEmpty(t, tokenSet.AccessToken)
+		assert.Equal(t, "Bearer", tokenSet.TokenType)
```

Comment:
Is this assert relevant for this test?

Reply:
It's not required but I left it in most assertions as a way of validating the response is returned as expected due to tokens not really being testable